### PR TITLE
strace and gdb backtrace

### PR DESCRIPTION
**Pull request by doniks**

### CODE REVIEW

File: porting/debug-build/debug-android-userspace.rst [hunk 1]
```diff
@@ -51,6 +51,22 @@ Even though Android logs do not normally end up in dmesg, early initialization o
...
+
+backtrace
+---------
+
+Another debugging technique is to investigate the backtrace when a program crashes. This is how you get a backtrace for the example of ``test_hwcomposer``::
+
+   EGL_PLATFORM=hwcomposer gdb test_hwcomposer
+
+This will start the interactive debugger ``gdb``. At the prompt of ``gdb`` you enter ``run``. Now the program is executed and you wait for it to crash. Then you enter ``bt full``. This will give you the full backtrace of what the program was trying to execute at the moment of the crash.
+
```
[the remark this answers is not in the document]
Comment (doniks) [reply porting/debug-build/debug-android-userspace.rst]: fiiiinally. sorry for the delay. Please recheck.

There is this "Debug Libhybris crash"-section-todo ... I wonder whether this is actually done now?!

[the remark this answers is not in the document]
Comment (doniks) [reply porting/debug-build/debug-android-userspace.rst]: Ping

File: porting/debug-build/debug-android-userspace.rst [hunk 2]
```diff
@@ -51,10 +51,42 @@ Even though Android logs do not normally end up in dmesg, early initialization o
 
    dmesg
 
+strace
+------
+
+For cases where the log files do not reveal sufficient detail, an ``strace`` can be helpful. This is how you get an strace for the example of ``test_hwcomposer``::
```
Comment (vanyasem) [inline porting/debug-build/debug-android-userspace.rst]: I am not really sure, is it "a strace" or "an strace"?

File: porting/debug-build/debug-android-userspace.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (doniks) [reply porting/debug-build/debug-android-userspace.rst]: pong

[the remark this answers is not in the document]
Comment (doniks) [reply porting/debug-build/debug-android-userspace.rst]: Cough

**Comment by doniks** [comment]
> @doniks, do you feel that the "debug libhybris" todo can go away with this?

not sure. that's why I left it in. @caspermeijn https://github.com/Halium/docs/commit/4c3af58424ccb718c45dd733b05a0850e09c8dc5 added it. But if no one really knows or says, we might as well remove it. No point keeping an item that's not actionable

**Comment by UniversalSuperBox** [comment]
I'd say remove it

**Comment by doniks** [comment]
sh*t. I just screwed up the PR

**Comment by doniks** [comment]
if at first you don't succeed, https://github.com/Halium/docs/pull/120

**Comment by mikroskeem** [comment]
@doniks you can always use `git rebase -i` and force push. :smile:

**Comment by doniks** [comment]
Yeah, so the branch of this pr was super old. I think I tried to update it by `git rdbase origin master` but then I ended up with changes all over the place like I was comitting every change of the last month again... I need to practice this more

**Comment by caspermeijn** [comment]
Fyi, I only added a newline in that commit :-)

doniks <notifications@github.com> schreef op 15 oktober 2018 18:37:02 CEST:

**Comment by doniks** [comment]
oh. indeed, I it see now, https://github.com/Halium/docs/commit/18607965bfec5502f19bec4bbbd70c63340e4bdf  a certain UniversalSuperBox :-P oh well

**Pull request closed without merging**